demos/search.js

```diff
@@ -0,0 +1,613 @@
+		"Desert Wheatear" : "Oenanthe deserti",
+		"White`s Thrush" : "Zoothera aurea",
+		"Siberian Thrush" : "Zoothera sibirica",
+		"Eyebrowed Thrush" : "Turdus obscurus",
+		"Dusky Thrush" : "Turdus eunomus",
+		"Black-throated Thrush" : "Turdus atrogularis",
+		"Pallas`s Grasshopper Warbler" : "Locustella certhiola",
+		"Spectacled Warbler" : "Sylvia conspicillata",
+		"Two-barred Warbler" : "Phylloscopus plumbeitarsus",
+		"Eastern Bonelli’s Warbler" : "Phylloscopus orientalis",
+		"Collared Flycatcher" : "Ficedula albicollis",
+		"Wallcreeper" : "Tichodroma muraria",
+		"Turkestan Shrike" : "Lanius phoenicuroides",
+		"Steppe Grey Shrike" : "Lanius pallidirostris",
+		"Spanish Sparrow" : "Passer hispaniolensis",
+		"Red-eyed Vireo" : "Vireo olivaceus",
+		"Myrtle Warbler" : "Dendroica coronata",
+		"White-crowned Sparrow" : "Zonotrichia leucophrys",
+		"White-throated Sparrow" : "Zonotrichia albicollis",
+		"Cretzschmar`s Bunting" : "Emberiza caesia",
+		"Chestnut Bunting" : "Emberiza rutila",
+		"Red-headed Bunting" : "Emberiza bruniceps",
+		"Black-headed Bunting" : "Emberiza melanocephala",
+		"Indigo Bunting" : "Passerina cyanea",
+		"Balearic Woodchat Shrike" : "Lanius senator badius",
+		"Demoiselle Crane" : "Grus virgo",
+		"Chough" : "Pyrrhocorax pyrrhocorax",
+		"Red-Billed Chough" : "Pyrrhocorax graculus",
+		"Elegant Tern" : "Sterna elegans",
+		"Chukar" : "Alectoris chukar",
+		"Yellow-Billed Cuckoo" : "Coccyzus americanus",
+		"American Sandwich Tern" : "Sterna sandvicensis acuflavida",
+		"Olive-Tree Warbler" : "Hippolais olivetorum",
+		"Eastern Olivaceous Warbler" : "Acrocephalus pallidus",
+		"Indian Cormorant" : "Phalacrocorax fuscicollis",
+		"Spur-Winged Lapwing" : "Vanellus spinosus",
+		"Yelkouan Shearwater" : "Puffinus yelkouan",
+		"Trumpeter Finch" : "Bucanetes githagineus",
+		"Red Grouse" : "Lagopus scoticus",
+		"Rock Ptarmigan" : "Lagopus mutus",
+		"Long-Tailed Cormorant" : "Phalacrocorax africanus",
+		"Double-crested Cormorant" : "Phalacrocorax auritus",
+		"Magnificent Frigatebird" : "Fregata magnificens",
+		"Naumann's Thrush" : "Turdus naumanni",
+		"Oriental Pratincole" : "Glareola maldivarum",
+		"Bufflehead" : "Bucephala albeola",
+		"Snowfinch" : "Montifrigilla nivalis",
+		"Ural owl" : "Strix uralensis",
+		"Spanish Wagtail" : "Motacilla iberiae",
+		"Song Sparrow" : "Melospiza melodia",
+		"Rock Bunting" : "Emberiza cia",
+		"Siberian Rubythroat" : "Luscinia calliope",
+		"Pallid Swift" : "Apus pallidus",
+		"Eurasian Pygmy Owl" : "Glaucidium passerinum",
+		"Madeira Little Shearwater" : "Puffinus baroli",
+		"House Finch" : "Carpodacus mexicanus",
+		"Green Heron" : "Butorides virescens",
+		"Solitary Sandpiper" : "Tringa solitaria",
+		"Heuglin's Gull" : "Larus heuglini"
+	};
+	function searchResponse( term ) {
+		term = term.toLowerCase();
+		var result = [];
+		for ( var key in database ) {
+			if ( key.toLowerCase().indexOf( term ) !== -1 ) {
```

Comment:
I was not able to find where this would be triggered in the demos. The four demo pages that include this script, set `minLength: 2`. 

Looking at the non-remote demos, it seems that several of them allow empty input to offer the first few values of the data source, which I presume is an intentional and useful UX feature. E.g. <https://jqueryui.com/autocomplete/#multiple> press down arrow to reveal the available programming languages.

I removed the check on the assumption that it was there only as a server protection for long responses. I imagine that if there is a demo in the future without `minLength: 2`, it would be expected that it can return values regardless of whether there is a length or not. I wouldn't write extra code now to make that work in the future, but it did seem strange to wrote extra code to specifically not make that work. Separation of concerns, etc, this is merely a data source function.

Reply:
OK, fair enough. 🙂